src/main/java/com/github/vkorobkov/jfixturescmd/CmdParser.java

```diff
@@ -0,0 +1,48 @@
+package com.github.vkorobkov.jfixturescmd;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.ParameterException;
+import com.github.vkorobkov.jfixtures.JFixtures;
+import com.github.vkorobkov.jfixtures.fluent.JFixturesResultImpl;
+import com.github.vkorobkov.jfixturescmd.utils.PropertiesReader;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+
+import java.lang.reflect.Method;
+
+@Slf4j
+final class CmdParser {
+    private final CmdArgs cmdArgs = CmdArgs.getInstance();
+
+    void parse(final String[] args) {
+        final JCommander jCommander = buildJCommander();
+
+        try {
+            jCommander.parse(args);
+
+            if (cmdArgs.isHelp()) {
+                jCommander.usage();
+            } else {
+                generateSql();
+            }
+        } catch (ParameterException e) {
+            log.error("Failed to parse command line arguments: " + e.getMessage());
+            jCommander.usage();
+        }
+    }
+
+    private JCommander buildJCommander() {
+        return JCommander.newBuilder()
+                    .addObject(cmdArgs)
+                    .programName(PropertiesReader.PROJECT_NAME)
+                    .build();
+    }
+
+    @SneakyThrows
+    private void generateSql() {
+        String methodName = String.valueOf(cmdArgs.getSqlType()).toLowerCase();
```

Comment:
Let's try to do things without reflection in a KISS way:

1. Define a method
```java
JFixturesResult chooseDialect(SqlTypes type, String src) {
  switch(type) {
    case POSTGRES:
      return JFixtures.postgres(src);
    // handle other cases
    default:
      throw new IlleagalStateException(type + " is not supported.");
  }
}
```
2. Call it from `generateSql()`

Besides that, I've created a ticket for moving the `SqlTypes` enum into core, but this refactoring should be done as a separate unit